bindgen/codegen/mod.rs

```diff
@@ -2470,17 +2527,130 @@ impl CodeGenerator for CompInfo {
         // NB: We can't use to_rust_ty here since for opaque types this tries to
         // use the specialization knowledge to generate a blob field.
         let ty_for_impl = quote! {
-            #canonical_ident #generics
+            #canonical_ident #impl_generics_params
         };
 
         if needs_clone_impl {
             result.push(quote! {
-                impl #generics Clone for #ty_for_impl {
+                impl #impl_generics_labels Clone for #ty_for_impl {
                     fn clone(&self) -> Self { *self }
                 }
             });
         }
 
+        if needs_flexarray_impl {
+            let prefix = ctx.trait_prefix();
```

Comment:
Can we move this out to its own function or such? I feel like that'd make it easier to reason about.

Reply:
Yeah, happy to do that. But the rest of the function is so enormous - do you think it should all be factored down?